[docs] Gh #5196 describe current limitations (#5428)
DevExpress/testcafe: docs/articles/documentation/reference/test-api/testcontroller/setnativedialoghandler.md

# t.setNativeDialogHandler Method

To handle native dialogs invoked during the test run, specify a handler function
with the `setNativeDialogHandler` method of the

> [test controller](README.md).

Reviewer: Let's rewrite this and say:

`To handle native dialogs invoked during the test run, use the `setNativeDialogHandler` method of the [test controller](README.md) to specify a handler function.`

Author: I don't like the structure `**To** do this, use this **to** do this`

Can we say:

> To handle native dialogs invoked during the test run, specify a handler function with the setNativeDialogHandler method of the [test controller](README.md).

Reviewer: I see, since these instructions are a little more complex (to the reader), we can go with your original suggestion. Thanks.

`text`   | String | Text of the dialog message.
`url`    | String | The URL of the page that invoked the dialog. Use it to determine if the dialog originated from the main window or an `<iframe>`.

> Once the handler is specified, it is triggered each time a native dialog appears in the test whether it originates from the main window or an `<iframe>`.

Reviewer: ```suggestion
Once the handler is specified, it is triggered each time a native dialog appears in the test - whether it originates from the main window or an `<iframe>`.
```

Author: Yes, the dialog

An attempt:
> Once specified, the handler fires each time a native dialog appears in the test. The dialog can originate from the main window or an `<iframe>`.

Reviewer: `Once specified, the handler fires each time a native dialog appears in the test. The dialog can originate from the main window or an <iframe>.`

Thank you. Please apply your new suggestion.

Author: Done

> You can specify a new handler at any moment by calling `t.setNativeDialogHandler` once again.

Reviewer: ```suggestion
You can call `t.setNativeDialogHandler` again to specify a new handler at any time.
```

If a native dialog appears when no handler is set, the test fails with an error.

> You can remove a dialog handler by passing `null` to the `t.setNativeDialogHandler` method.

Reviewer: ```suggestion
To remove a dialog handler, pass `null` to the `t.setNativeDialogHandler` method.
```

> If a dialog appears on page load, start the test from a different page, add a handler and proceed to the page with a [navigateTo](./navigateto.md) action, per the [example](#handle-a-dialog-invoked-on-page-load).

Reviewer: ```suggestion
If a dialog appears on page load, start the test from a different page, add a handler and proceed to the page with the [navigateTo](./navigateto.md) action, as shown in the the following example: [Handle a Dialog Invoked on Page Load](#handle-a-dialog-invoked-on-page-load).
```

> > The handler is executed on the client side, so you cannot use Node.js API in the handler.

Reviewer: ```suggestion
> The handler is executed on the client side, so you cannot use the Node.js API in the handler.
```

> You can control how a dialog is handled with handler's return values.

Reviewer: ```suggestion
You can use the handler's return values to control how the dialog is handled.
```

});
```

### Handle a Dialog Invoked on Page Load

> As soon as this example page loads, an alert blocks it. To set a handler for this alert, you can start the test from an `about:blank` page, add a handler and proceed to the page with an `navigateTo` action.

Author: I like your version way more.

Also, should I add a warning about relative paths and `about:blank` here?

> Important! Use absolute file paths to navigate from `about:blank`.

> As soon as this example page loads, an alert blocks it. To set a handler for this alert, you can start the test from an `about:blank` page, add a handler and proceed to the page with an `navigateTo` action.

Reviewer 2: Yes, in this section it's good to add it, thanks!

});
```

### Handle a Dialog Invoked on Page Load

> When the page in this example loads, it displays an alert dialog. To set a handler for this alert, you can start the test from an `about:blank` page, add a handler and proceed to the page with an `navigateTo` action.

Reviewer: ```suggestion
When the page in this example loads, it displays an alert dialog. To set a handler for this alert, you can start the test from an `about:blank` page, add a handler and proceed to the page with the `navigateTo` action.
```
